src/snowflake/snowpark/mock/mock_connection.py

```diff
@@ -64,6 +71,63 @@ def _build_target_path(stage_location: str, dest_prefix: str = "") -> str:
 
 
 class MockServerConnection:
+    class TableRegistry:
+        # Table registry. TODO: move to datastore
+        def __init__(self, conn: "MockServerConnection") -> None:
+            self.table_registry = {}
+            self.conn = conn
+
+        def get_fully_qualified_name(self, name: Union[str, Iterable[str]]) -> str:
+            def uppercase_and_enquote_if_not_quoted(string):
+                if string[0] == '"' and string[-1] == '"':
```

Comment:
should this check be:
```suggestion
                if len(string) >= 2 and string[0] == '"' and string[-1] == '"':
```

Reply:
I see, then there's also the case when there are quotes inside the quotes that need to be escaped.